apps/api/src/donations/donations.controller.ts

```diff
@@ -31,4 +35,62 @@ export class DonationsController {
   findRecurringPrices() {
     return this.paymentsService.listPrices('recurring')
   }
+
+  @Get('list')
+  @Public()
+  findAll() {
+    return this.paymentsService.listDonations()
+  }
+
+  @Get(':id')
+  @Public()
+  findOne(@Param('id') id: string) {
+    return this.paymentsService.getDonationById(id)
+  }
+
+  @Post('create-payment')
+  create(
+    @AuthenticatedUser()
+    user: KeycloakTokenParsed,
+    @Body()
+    createPaymentDto: CreatePaymentDto,
+    
+  ) {
+    if (!user) {
+      throw new UnauthorizedException()
+    }
+
+    return this.paymentsService.create(createPaymentDto, user)
+  }
+
+  @Patch(':id')
```

Comment:
We need to restrict the access to update and delete of donations to team members with:

```tsx
@Roles({
  roles: [RealmViewSupporters.role, ViewSupporters.role],
  mode: RoleMatchingMode.ANY,
})
```

Reply:
@angelina-7 I've done this comment myself on top of your branch.